### DISCUSSION TITLE
Is it possible to make an exact replica of a system design from GitHub?<title>

### DISCUSSION THREAD

**Discussion by ruru-m07** [community/community#110788] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

# GitHub-like Application with GCS Object Versioning

I'm currently working on an application similar to GitHub, and I'm using GCS object versioning. I've made progress in handling different versions of files, but I still need more help for the next steps.

So far, I've completed some minor tasks. You can visit my website at [Cocola Tech](https://www.cocola.tech/new) and create a free account. Please note that this project is still under development, so some buttons are only for viewing purposes. 

After creating your account, navigate to the `/new` page to create a new repository. You'll then be redirected to `/:yourusername/:yourreponame`. Here, you'll find a button labeled 'add files'. Click on it to upload some files and folders. Make sure to include a `README.md` file. 

Once you've uploaded the files, create a commit. Then, go back to 'add files' and upload an updated `README.md` file. You'll be able to see the updated files on the repository page.

# login page
![image](https://github.com/community/community/assets/142723369/8924243d-f5d9-44f8-92a3-6253f4fb4a9b)

# `/new` page
![image](https://github.com/community/community/assets/142723369/1fc285d7-7dfe-4446-91de-10ba34626cf6)

# upload page
![image](https://github.com/community/community/assets/142723369/f6b18602-7436-441a-9086-2b4ede76a84b)

# `/:yourusername/:yourreponame` page
![image](https://github.com/community/community/assets/142723369/3f2c2285-99e6-4280-bb34-0920549e6386)

**Comment by timmawv** [comment]


**Reply by ruru-m07** [reply]
# Seeking Further Assistance with System Design

Thank you for your response. However, I still require assistance in designing a better and more scalable system. If you could take a look at my project, it would help you understand what I'm aiming for and how you can provide more targeted assistance.

**Reply by grandmas-favorite** [reply]
Hi @timmawv, 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
